### PR TITLE
openPMD-api 0.15.1

### PR DESCRIPTION
**Pull request by ax3l**
Update WarpX to use openPMD-api `0.15.1`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -116,7 +116,7 @@ option(WarpX_IPO                                "Compile WarpX with interprocedu
 #   this is temporary and we don't want to set it long term
 #   mainly to work-around a global setting in openPMD-api <= 0.14.5 that changes the
 #   default in superbuilds by accident
-if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
+if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)  # update me
```
Comment (ax3l) [inline CMakeLists.txt]: update me

File: Tools/machines/summit-olcf/summit_warpx.profile.example [hunk 2]
```diff
@@ -28,7 +28,7 @@ module load hdf5/1.10.7
 # optional: for openPMD support (GNUmake only)
 #module load ums
 #module load ums-aph114
-#module load openpmd-api/0.14.2
+#module load openpmd-api/0.15.0
```
Comment (ax3l) [inline Tools/machines/summit-olcf/summit_warpx.profile.example]: Deploy me (optional, GNUmake only)

File: CMakeLists.txt (hunk 1 above)
Comment (ax3l) [reply CMakeLists.txt]: To do: also need to update the default in `openPMD.cmake` from shared to static.

Comment (ax3l) [reply CMakeLists.txt]: done

**Review by RemiLehe — APPROVED** [review]
Thanks! ✨

**Pull request merged**